fairroot/parbase/FairDetParAsciiFileIo.cxx

```diff
@@ -42,24 +45,26 @@ FairDetParAsciiFileIo::FairDetParAsciiFileIo(std::fstream* f)
 Bool_t FairDetParAsciiFileIo::findContainer(const Text_t* name)
 {
     // searches the container in the file
-    const Int_t maxbuf = 4000;
-    Text_t buf[maxbuf];
-    Text_t buf2[maxbuf];
-    sprintf(buf2, "%s%s%s", "[", name, "]");
-    // cout << " buf2 " <<  buf2 << endl;
+    const auto searchValue = fmt::format("[{}]", name);
+    const auto searchValueLength = searchValue.size();
     pFile->clear();
     pFile->seekg(0, ios::beg);
-    while (!pFile->eof()) {
-        pFile->getline(buf, maxbuf);
-        if (buf[0] != '[') {
+    // looking for lines that start with searchValue
+    for (std::string line; std::getline(*pFile, line);) {
+        // TODO: C++20
+        // if (line.starts_with(searchValue)) {
+        //     return true;
+        // }
+
+        if (line.size() < searchValueLength) {
             continue;
         }
-        // cout << " buf: " <<  buf << endl;
-        if (!strncmp(buf, buf2, strlen(buf2))) {
-            break;
+        if (const auto linePrefix = std::string_view(line).substr(0, searchValueLength);
+            linePrefix == searchValue) {
+            return true;
         }
     }
-    return !pFile->eof();
+    return false;
 }
```

Comment:
@karabowi @ChristianTackeGSI Please double check this. Not sure if we have test coverage.

Reply:
Looks good to me. I still wonder why this thing needs to check only the prefix. But I haven't investigated the code around.

Reply:
Na, if you cannot read it, we should put it into a function `startsWith()`

Reply:
How does the commented C++20 code read to you compared to `!strncmp(buf, buf2, ...)`?

Reply:
I said I need time to digest it, not that I cannot read it.

Reply:
Sure, but you made the argument, it is less readable to you. And my main point of changing this code is to make it more readable ;)